Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -1785,43 +1785,904 @@ public void When_asserting_string_does_not_end_with_equivalent_of_a_value_and_ac
+            Action act = () => actual.Should().Contain(expectedSubstring).AtMost.Times(expectedTimes);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_string_containment_at_most_is_asserted_and_actual_value_contains_the_expected_string_but_not_at_most_expected_times_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            string actual = "ABCDEBCDF";
+            string expectedSubstring = "BCD";
+            int expectedTimes = 1;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => actual.Should().Contain(expectedSubstring).AtMost.Times(expectedTimes, "that is {0}", "required");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().Throw<XunitException>().WithMessage(
+#if NETCOREAPP1_1
+                "Expected string \"ABCDEBCDF\" to contain \"BCD\" at most 1 time because that is required, but found 2 times.");
+#else
+                "Expected actual \"ABCDEBCDF\" to contain \"BCD\" at most 1 time because that is required, but found 2 times.");
+#endif
+        }
+
+        [Fact]
+        public void When_string_containment_at_most_once_is_asserted_and_actual_value_does_not_contain_the_expected_string_it_should_throw_earlier()
         {
             //-----------------------------------------------------------------------------------------------------------
             // Arrange
             //-----------------------------------------------------------------------------------------------------------
             string actual = "ABCDEF";
+            string expectedSubstring = "XYS";
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => actual.Should().Contain(expectedSubstring).AtMost.Once("that is {0}", "required");
```

Comment:
❌ This is a prime example of why I don't like this augmentation of `Contain`. 
`ABCDEF` contains `XYS` 0 times, which is less than 1, but the test fails.

Reply:
You're right, that was my assumption.
I think that you as maintainers should decide about API.

I see following solutions:
1. Leave it as is
2. Replace the current `Contain` by something like that `actual.Should().Contain("Fred").Any`
3. Expand `Contain` in different way, e.g. as you suggested in #818 
4. Introduce new method, but I don't know how to named it.

Reply:
@dennisdoomen Here's my beef with this API.

Reply:
Hi,

I think that I have found the way to fix this without breaking changes, by using `new AssertionScope()` instead of `Execute.Assertion`.
Please give me some additional time, I will update the PR probably on Wednesday.

Reply:
The problem they were facing had to do with the word _repeated_. We don't have this issue with this API. It reads very fluently IMHO.

Reply:
@dennisdoomen I agree that it reads fluently, but the result of
```c#
"A".Should().Contain("B").AtMost.Once();
```
would fail and be logically inconsistent with the written assertion.

Reply:
Yeah, either we make sure our API does not support that, or we reconsider the API